src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabService.java

```diff
@@ -333,7 +333,7 @@ public void createProjectForExercise(ProgrammingExercise programmingExercise) th
         for (final var instructor : instructors) {
```

Comment:
I think that code assumes that a user can only have one role for the repository.
This causes #3413 (at least I think so ..)

I would suggest to modify the code in a way that for each user the most powerful role (that the user has) will be used.
E.g. is a user is tutor and editor, the user gets the rights of an editor.

Reply:
You're right, I changed the ordering of the calls so that the lowest access level is applied first and the highest level last. If the tutor is a teaching assistant and an editor, the editor role will be applied

Reply:
Seems to be good. I think that should work. In order to reduce the amount of calls to gitlab, we should maybe just search for the most privileged role and sent one request per user. But that would be just an optimization.

Reply:
@ivanchimeno what's your opinion on this optimization? Is that in the scope of this PR ?

Reply:
@dfuchss I optimized the code, let me know how you find it now